lib/globalize/active_record.rb

```diff
@@ -32,12 +32,14 @@ def locale=(locale); write_attribute(:locale, locale.to_s); end
     end
 
     module ActMacro
+      ThreadLocalKey = :globalize2_locale
+
       def locale
-        (defined?(@@locale) && @@locale)
+        Thread.current.key?(ThreadLocalKey) && Thread.current[ThreadLocalKey]
```

Comment:
oh so this is what `Thread` does https://www.rubytapas.com/2016/11/20/ruby-thread-local-variables/

as a side note I can't remember the reason why i did this whole branch lol 😬